cookbooks/aws-parallelcluster-environment/libraries/directory_service_utils.rb

```diff
@@ -0,0 +1,24 @@
+#
+# Parse parameter DomainReadOnlyUser: cn=ReadOnlyUser,ou=Users,ou=CORP,dc=corp,dc=sirena,dc=com
+# and return the actual Name of the ReadOnly user
+def _get_cn_subparam(param)
+  tokens = param.split(",") unless param.blank?
+  tokens.each do |token|
+    return token if token.include? "cn"
+  end
+  ""
+end
+
+def domain_service_read_only_user_name(param)
+  cn_token = _get_cn_subparam(param)
+  if cn_token.blank?
+    Chef::Log.info("Failed to retrieve the ReadOnlyUser from #{param}")
+    name = 'ReadOnlyUser'
+    Chef::Log.info("Falling back to the default UserName #{name}")
```

Comment:
We do not have an actual ReadOnly user default naming.

Reply:
Got it ..we can keep this anyway ..since if we enter this branch the configuration is broken and the getent request will fail